tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -5507,6 +5507,70 @@ public function dataArrayFunctions(): array
 				'\'foo\'',
 				'$poppedFoo',
 			],
+			[
+				'int',
+				'array_rand([1, 2])',
+			],
+			[
+				'string',
+				'array_rand(["1", "2"])',
```

Comment:
Maybe i'm misunderstanding something here, but `array_rand` returns a random **key** right?

So the return type here would a int, as this array is a list:
https://3v4l.org/H1hqZ